managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1536,7 +1537,18 @@ void asyncReadEntries(OpReadEntry opReadEntry) {
 
             LedgerInfo info = ledgers.get(ledgerId);
             CompletableFuture<ReadHandle> openFuture = new CompletableFuture<>();
-            if (info != null && info.hasOffloadContext() && info.getOffloadContext().getComplete()) {
+
+            if (config.getLedgerOffloader() != null
+                    && config.getLedgerOffloader().getOffloadPolicies() != null
+                    && config.getLedgerOffloader().getOffloadPolicies()
```

Comment:
The `OffloadPolicies` you get from `config` may not be updated as namespace or topic policy update.

Reply:
> The `OffloadPolicies` you get from `config` may not be updated as namespace or topic policy update.

Good question, I will try to figure it out

Reply:
@hangc0276 
I think 
`org.apache.pulsar.broker.admin.v2.PersistentTopics#setOffloadPolicies`
`org.apache.pulsar.broker.admin.impl.PersistentTopicsBase#internalSetOffloadPolicies`
`org.apache.pulsar.broker.admin.impl.PersistentTopicsBase#internalUpdateOffloadPolicies`
`org.apache.bookkeeper.mledger.impl.ManagedLedgerImpl#setConfig`

and 

`org.apache.pulsar.zookeeper.ZooKeeperCache#process(org.apache.zookeeper.WatchedEvent, org.apache.pulsar.zookeeper.ZooKeeperCache.CacheUpdater<T>)`
`org.apache.pulsar.zookeeper.ZooKeeperChildrenCache#reloadCache`
`org.apache.pulsar.broker.service.BrokerService#onUpdate`
`org.apache.pulsar.broker.service.persistent.PersistentTopic#onPoliciesUpdate`
`org.apache.pulsar.broker.service.persistent.PersistentTopic#checkPersistencePolicies`
`org.apache.bookkeeper.mledger.impl.ManagedLedgerImpl#setConfig`

are assurable chains to keep config up to date